code/game/objects/items/weapons/shields.dm

```diff
@@ -98,27 +98,27 @@
 	if(user.skills.getRating("engineer") < SKILL_ENGINEER_METAL)
 		user.visible_message(span_notice("[user] fumbles around figuring out how to repair [src]."),
 		span_notice("You fumble around figuring out how to repair [src]."))
-		var/fumbling_time = 3 SECONDS * ( SKILL_ENGINEER_METAL - user.skills.getRating("engineer") )
+		var/fumbling_time = 4 SECONDS * ( SKILL_ENGINEER_METAL - user.skills.getRating("engineer") )
 		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_BUILD))
 			return TRUE
 
-	user.visible_message(span_notice("[user] begins repairing damage to [src]."),
-	span_notice("You begin repairing the damage to [src]."))
+	user.visible_message(span_notice("[user] begins repairing [src]."),
+	span_notice("You begin repairing [src]."))
 	playsound(loc, 'sound/items/welder2.ogg', 25, TRUE)
 
-	if(!do_after(user, 3 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY))
+	if(!do_after(user, 4 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY))
 		return TRUE
 
 	if(obj_integrity <= (max_integrity - integrity_failure) * 0.2 || obj_integrity == max_integrity)
 		return TRUE
 
 	if(!WT.remove_fuel(2, user))
-		to_chat(user, span_warning("Not enough fuel to finish the task."))
+		to_chat(user, span_warning("Not enough fuel to finish the repairs."))
 		return TRUE
 
-	user.visible_message(span_notice("[user] repairs some damage on [src]."),
-	span_notice("You repair [src]."))
-	repair_damage(40)
+	user.visible_message(span_notice("[user] finishes repairing [src]."),
+	span_notice("You finish repairing [src]."))
+	repair_damage((src.max_integrity-src.integrity_failure) * 0.2)
```

Comment:
Bear in mind, what you're doing here is essentially spending four seconds to heal a human's extra healthpool by 60 for no resources. Scaling off the difference between max int and failure point is also kind of weird for this? And you don't need the srcs

Reply:
Repair damage is scaled off failure point since thats when it shows up as broken.

Reply:
Where do you not need the srcs? Please point that out.

Reply:
This line, no need of any src

Reply:
> 

Number of the line?